### PR TITLE
[Approaches]: Raindrops

### PR DESCRIPTION
**Pull request by glaxxie**
WIP
quick benchmark show `switch` statement is the fastest

### CODE REVIEW

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    $sounds.Count -eq 0 ? "$Number" : -join $sounds
+}
+```
+
+For more information, check the [`hashtable` approach][approach-ordered-hashtable].
+
+
+## Which approach to use?
+
+- The if statements approach is simple and straight to the point, however it has drawback from needing to create new string every time you do string concatenation.
```
Comment (ErikSchierboom) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
- The if statements approach is simple and straight to the point, however it has the drawback of needing to create a new string every time you do string concatenation.
```

**Review by ErikSchierboom — APPROVED** [review]
This is great content and I feel like all three approaches are definitely different enough to warrant separate approaches. 

My only general suggestion would be to add some lines of code to the explanations, to help prevent the student from having to scroll up to see what the code was that was being explained. Here is an example: https://exercism.org/tracks/csharp/exercises/hamming/approaches/linq

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    $sounds.Length -eq 0 ? "$Number" : $sounds
+}
+```
+
+For more information, check the [`if` statements approach][approach-if-statements].
+
+
+## Approach: Using `switch` statement
+
+Structurely, this approach is very similar to the `if` statements approach, however the `switch` statement lead to a cleaner syntax and a straight forward path to the result.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
Structurally, this approach is very similar to the `if` statements approach, however the `switch` statement leads to a cleaner syntax and a straight forward path to the result.
```

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+The goal of the exercise is to check if the input number is divisible by 3, 5 and 7, or maybe none of those.
+For determining that, you can use the `modulus` operator (`%`).
+
+*Note* : The order of testing is crucial in this exercise to arrive to the correct result.
+
+## Approach: Using `if` statements
+
+This is the most common approach to solve this exercise.
+The result string being built up by a succession of `if` statements to check of the input can be divided cleanly by 3, 5 and 7.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
The result string is built up by a succession of `if` statements checking if the input can be divided cleanly by 3, 5, and 7.
```

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# Introduction
+
+Raindrops exercise is a twist of the popular `FizzBuzz` problem.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
Raindrops exercise is a twist on the popular `FizzBuzz` problem.
```

Also perhaps there should be a link to what FizzBuzz is if the reader doesn't know the term.

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+There are a couple ways to solve it idiomatic in PowerShell.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
There are a couple ways to solve it idiomatically in PowerShell.
```

File: exercises/practice/raindrops/.approaches/introduction.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    $sounds = foreach ($key in $table.Keys) {
+        if ($Number % $key -eq 0) {
+            $table.$key
+        }
+    }
+    $sounds.Count -eq 0 ? "$Number" : -join $sounds
+}
+```
+
+For more information, check the [`hashtable` approach][approach-ordered-hashtable].
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md]:
```suggestion
For more information, check the [`ordered hashtable` approach][approach-ordered-hashtable].
```

File: exercises/practice/raindrops/.approaches/ordered-hashtable/content.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    $sounds = foreach ($key in $table.Keys) {
+        if ($Number % $key -eq 0) {
+            $table.$key
+        }
+    }
+    $sounds.Count -eq 0 ? "$Number" : -join $sounds
+}
+```
+
+This approach loop through an ordered `hashtable`'s keys.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/ordered-hashtable/content.md]:
```suggestion
This approach loops through an `ordered hashtable`'s keys.
```

File: exercises/practice/raindrops/.approaches/ordered-hashtable/content.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+If the input number is divisible against a key, then use that key to look up the specific string in the `hashtable`.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/ordered-hashtable/content.md]:
```suggestion
If the input number is divisible by a key, retrieve the string associated with the key from the hashtable.
```

File: exercises/practice/raindrops/.approaches/ordered-hashtable/content.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+The loop can be assign into a variable to represent the final collection.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/ordered-hashtable/content.md]:
```suggestion
The loop can be assigned to a variable to represent the final collection of sounds.
```

File: exercises/practice/raindrops/.approaches/ordered-hashtable/content.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+If the collect is empty at the end, return the original number as string.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/ordered-hashtable/content.md]:
```suggestion
If the collection is empty at the end, return the original number as string.
```

File: exercises/practice/raindrops/.approaches/switch-statement/content.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        ($Number % 7) {"Plong"}
+        Default {$Number}
+    }
+    -join $sounds
+}
+```
+
+This approach is similar to the `if` statements.
+The value `0` is used to match if the input number is divisible for `3`, `5` and `7`.
+And if none of those give a match then the default will be the original number.
```
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/switch-statement/content.md]:
```suggestion
And if none of those conditions are true, the default value will be the original number.
```

File: exercises/practice/raindrops/.approaches/introduction.md
Comment (BNAndras) [inline exercises/practice/raindrops/.approaches/introduction.md:1]: Don't we need the links for the approaches at the bottom for the Markdown to work? See https://raw.githubusercontent.com/exercism/racket/main/exercises/practice/leap/.approaches/introduction.md

Comment (glaxxie) [reply exercises/practice/raindrops/.approaches/introduction.md:1]: thank you

**Comment by glaxxie** [comment]
@ErikSchierboom can you take a quick look at the updated content, and give the thumb up before I merge. Thanks.

**Review by ErikSchierboom — APPROVED** [review]
This is great!

**Pull request merged**